apps/emqx/src/emqx_broker_helper.erl

```diff
@@ -268,43 +273,52 @@ collect_messages(Reg, Down) ->
 %% There is no other message sent to this process, so the
 %% 'receive' should not have to scan the mailbox.
 collect_messages(Reg, Down, 0) ->
-    {Reg, Down};
+    {lists:reverse(Reg), lists:reverse(Down)};
```

Comment:
Q: Why do we need to reverse the lists now?

Reply:
the tests asserts order, but in practice, it's unlikely to matter.